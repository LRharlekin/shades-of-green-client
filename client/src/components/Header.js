import React from 'react'
import '../App.css';

const Header = (props) => {  
  return (
    <div className="Header">
      <h1 data-testid="h1tag" className="HeadLine">{ props.text }</h1>
      {/* <img src={process.env.PUBLIC_URL + 'favicon-32.png'} alt="logo" /> */}
    </div>
  );
}

Header.defaultProps = {
  text: "Shades of Green"
}

export { Header }