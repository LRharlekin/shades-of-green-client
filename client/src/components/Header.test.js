import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // Correct import
import faker from "faker" // just for fun

import { Header } from './Header'; // using named import instead of import Header from ...

describe('Header.js', () => {
  afterEach(cleanup);
  
  it("renders", () => {
    const { asFragment } = render(<Header />);
    // snapshot test:
    expect(asFragment()).toMatchSnapshot();
  });
  
  it("uses custom text if specified", () => {
    const title = faker.company.bs()
    const { getByTestId } = render(<Header text={ title } />);
  
    expect(getByTestId("h1tag")).toHaveTextContent(title);
    expect(getByTestId("h1tag")).toHaveClass("HeadLine");
  })
  
  it("uses Shades of Green if no custom text is passed", () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId("h1tag")).toHaveTextContent("Shades of Green");
  })
})
