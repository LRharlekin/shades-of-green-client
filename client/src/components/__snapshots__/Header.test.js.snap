// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.js renders 1`] = `
<DocumentFragment>
  <div
    class="Header"
  >
    <h1
      class="HeadLine"
      data-testid="h1tag"
    >
      Shades of Green
    </h1>
  </div>
</DocumentFragment>
`;
